### DISCUSSION TITLE
Enter key for an entry

### DISCUSSION THREAD

**Discussion by Valenterie** [TomSchimansky/CustomTkinter#239] [Ideas]
Hey everyone ! I'm currently trying to do a todo list on tkinter and CTk and I'm having an issue : 
I want to add an item in my list with the CTkEntry by hitting the enter key, unfortunately, for this to work, I have to be in a "while True:" loop and it's not possible since it would break the rest of my code, here's my code : 

![code](https://user-images.githubusercontent.com/87931771/177775726-9d258bc4-12a0-4a84-877d-2d3246840939.png)

What's important is the beginning and the end, the "enter" variable is just if i hit the enter key, but it's not the good solution since it's not looking everytime if i hit the key but only on the first frame when i run the code, is there a solution to make it work ? i searched in "customtkinter_entry.py" and "customtkinter_settings.py" but it seems like there is no other way whatsoever.

**Comment by TomSchimansky** [comment]
You create a callback function and bind it to the entry like the following:
```python
import customtkinter

app = customtkinter.CTk()

def add_element_callback(event):
    print("add element:", entry.get())

entry = customtkinter.CTkEntry(app)
entry.pack(pady=20, padx=20)
entry.bind("<Return>", add_element_callback)

app.mainloop()
```

**Reply by Valenterie** [reply]
Hey ! Thanks a lot for that solution, didn't know about the bind function ! Gonna explore that, keep the good work !

**Reply by Valenterie** [reply]
Welcome back, I have another problem, so I can click on the entry, enter a task that goes into my Listbox, but unfortunately it doesn't work if I try to do it again
